compiler/src/iree/compiler/Dialect/Flow/TransformExtensions/FlowExtensions.cpp

```diff
@@ -576,12 +576,45 @@ transform_dialect::MovePrecedingOpIntoDispatchRegionOp::apply(
 //
 // All operands of the target are replaced with values defined inside of the
 // dispatch region when possible.
+static FailureOr<Operation *> cloneSucceedingOpIntoDispatchRegion(
+    RewriterBase &rewriter, Operation *target,
+    Flow::DispatchRegionOp regionOp) {
+  assert(regionOp->isBeforeInBlock(target) &&
+         "expected that region op comes first");
+  Block &body = regionOp.getBody().front();
+
+  // Gather all uses of `target`.
+  SmallVector<OpOperand *> usesOutsideOfRegion;
+  for (OpOperand &use : target->getUses()) usesOutsideOfRegion.push_back(&use);
+
+  // Clone op into dispatch region.
+  auto returnOp = cast<Flow::ReturnOp>(body.getTerminator());
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(returnOp);
+  Operation *newTargetOp = rewriter.clone(*target);
+
+  // Replace all operands that are results of the regionOp.
```

Comment:
A little IR example would help here, this is non-standard.

Reply:
Added example to the function docs.